tests/module_cpp20/CMakeLists.txt

```diff
@@ -2,11 +2,21 @@ cmake_minimum_required(VERSION 3.28)
 
 project(json_test CXX)
 
-add_executable(json_test)
+set(nlohmann_json_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../build/modules)
 
-target_sources(json_test
-    PRIVATE main.cpp
-    PUBLIC FILE_SET cxx_modules TYPE CXX_MODULES FILES json.cpp)
+set(NLOHMANN_JSON_BUILD_MODULES ON CACHE BOOL "Enable nlohmann-json module support")
+
+add_subdirectory(../.. build)
```

Comment:
Please use `CMAKE_SOURCE_DIR` instead of relative paths. And again, check if `build` makes sense here.